hail/python/hail/context.py

```diff
@@ -381,7 +392,106 @@ def tmp_dir():
     return Env.hc()._tmpdir
 
 
-def current_backend():
+class _TemporaryFilenameManager:
+    def __init__(self, fs: FS, name: str):
+        self.fs = fs
+        self.name = name
+
+    def __enter__(self):
+        return self.name
+
+    def __exit__(self, type, value, traceback):
+        return self.fs.remove(self.name)
+
+
+def TemporaryFilename(*,
```

Comment:
Why not make this a class which is a context manager?

Reply:
Yeah, I guess I don't see why that code can't be in the `__init__` of the wrapper.  Well, doesn't really matter, your choice.